workflows/data_pipelines/elasticsearch/data_enrichment.py

```diff
@@ -153,14 +155,39 @@ def is_ess(est_ess_france, ess_insee):
 
 # Service public
 def is_service_public(nature_juridique_unite_legale, siren):
-    if (
-        nature_juridique_unite_legale
-        and nature_juridique_unite_legale.startswith(("4", "71", "72", "73", "74"))
-    ) or siren == "320252489":
-        return True
-    else:
+    """
+    Determine if a given entity is classified as a public service.
+
+    Parameters:
+    - nature_juridique_unite_legale (str): The legal nature of the entity.
+    - siren (str): The SIREN number of the entity.
+
+    Returns:
+    - bool: True if the entity is classified as a public service,
+            False otherwise. Exceptions include:
+            - BPI France (SIREN: 320252489) and URSSAF are considered public.
+            - RATP (SIREN: 775663438) is explicitly excluded.
+    """
+
+    # Exclude RATP
+    if siren == "775663438":
         return False
 
+    # Define valid prefixes for public service
+    valid_prefixes = {"4", "71", "72", "73", "74"}
+
+    # Check if the entity is classified as a public service
+    is_public = (
+        (
+            nature_juridique_unite_legale
+            and nature_juridique_unite_legale.startswith(tuple(valid_prefixes))
+        )
+        or siren == "320252489"  # BPI France
+        or siren in urssaf_siren_numbers  # Check against URSSAF SIREN numbers
```

Comment:
Why not using a list whitelisted siren that merge both BPI and Urssaf ? A simple hash with names in comments would work perfectly